core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -488,6 +489,18 @@ public SelfT withCloudSecureConnectBundle(@NonNull Path cloudConfigPath) {
     return self;
   }
 
+  /**
+   * Registers a CodecRegistry to use for the session.
+   *
+   * <p>When both this and {@link #addTypeCodecs(TypeCodec[])} are called, the added type codecs
+   * will be registered on the provided CodecRegistry.
+   */
+  @NonNull
+  public SelfT withCodecRegistry(@Nullable MutableCodecRegistry codecRegistry) {
```

Comment:
I change the parameter from NonNull to Nullable. This would allow someone to clear the registry on a builder where it was configured previously. Not very likely but the implementation allows it, so why not.

(there are other attributes on the builder that don't follow this pattern, I think it's an oversight on our side, I'll take a look)